### DISCUSSION TITLE
License

### DISCUSSION THREAD

**Discussion by caniko** [pydantic/pydantic#4257] [General]
Why did @samuelcolvin pick the MIT license? I'd say at least the Apache 2.0 license to protect the work from any possible issue in the future.

**If it was up to me**
Shouldn't a GPL license be better here? It will ensure that the project can't have derivatives on the proprietary side without that derivative work being closed. Also, was LGPL or BSD ever considered?

**Comment by samuelcolvin** [comment]
Lots of projects are dual MIT / Apache 2. What practical difference does that have?

I've recently acquired the trademark for "pydantic", though I'm not really sure if it could be effective at stopping people trying to use pydantic's brand good will for their own ends and thereby causing confusion.

I'm happy for pydantic to be used in products, and most of its use is in the cloud where GPL wouldn't actually have an effect anyway AFAIK.

The one thing I have thought about is licencing the docs in a different way to stop others hosting elsewhere.

I'm short I probably need to speak to a lawyer who has experience of this exact matter. Any suggestions?

**Reply by caniko** [reply]
I would speak to a lawyer.

I always thought that Apache 2.0 is a subsumption of the MIT license. Apache 2.0 is more explicit when it comes to protecting the copyright from bad actors, the list is longer, I just don't remember the exact details.

I also agree that companies shouldn't make their entire work open-source! Hence, my question for the consideration of LGPL and BSD.

GPL-like licenses are nice because it forces derivative work to be open-source, which can be added back into the project. Might not be relevant today, but years into the future; best to be on the safe side. [DeepLabCut](https://github.com/DeepLabCut/DeepLabCut/blob/master/LICENSE) uses the LGPL 3.0, I think it is a good fit for this project. LGPL  [Wikipedia article](https://en.m.wikipedia.org/wiki/GNU_Lesser_General_Public_License).

Regarding the docs. If you have the LGPL on the docs also, the publisher would have to disclose that they copied it from this repository.

**Reply by samuelcolvin** [reply]
Problems are:
1. in the end, since most use of pydantic is in the cloud and companies using it don't need to distribute the code, they can really do what they want, no license is going to stop them, particularly as it'll be clear to them that I don't have the firepower to enforce it
2. The letters "GPL" scare people - everyone knows "MIT is fine"

Then again, they also know "Apache 2 is fine" I think. We could probably migrate to Apache 2, through I imagine it would cause a bit of trouble and churn, the question is, what is the point?

---

On the docs, I'm not even clear on the process to change the license. Can I just put:
* a notice in the readme
* a notice in the current license file
* a license in the root of the `docs` directory

Or, do I need to have a separate PR, which will make updates which also change the docs much more complicated?

**Reply by caniko** [reply]
This is not advice, just my opinion @samuelcolvin 

> Problems are:

I understand, but should you make a decision based on what might be _scary_ to a few individuals, or what is best for the project? Nevertheless, not even Python is LGPL. That is why I agree that LGPL doesn't fit for the package license; it wouldn't fit the culture and would be more work than necessary with respect to your firepower observation.

The LGPL is probably more fitting to the documentation.

> Then again, they also know "Apache 2 is fine" I think. We could probably migrate to Apache 2, through I imagine it would cause a bit of trouble and churn, the question is, what is the point?

I am under the opinion that Apache 2.0 protects the project more, I might be wrong. Switching should be easy, MIT is compatible with Apache 2.0. It wouldn't have any retroactive effect anyway. I think this should be done to protect the project.

**Solution**
I think that you can have a separate license for the docs altogether. Just disclose this in the root `LICENSE` file. Something like:
```
Different components of the project are licensed differently after <version>:
The documentation is licensed under the LGPL v3.0 or later<link to LICENSE file in the documentation>
The Pydantic package is licensed under the Apache License v2.0<link to LICENSE file in the package>
```

> On the docs, I'm not even clear on the process to change the license. Can I just put:

To be pedantic, I don't think you should make your own license, much better to use a reviewed license. LGPL is compatible with Apache, [source](https://softwareengineering.stackexchange.com/questions/219397/how-to-use-apache-license-in-my-project-which-will-be-lgpl).

> Or, do I need to have a separate PR, which will make updates which also change the docs much more complicated?

This won't make any changes to the docs unless you want to add a notice under the docs, which does make sense. I don't see an issue to do all this under one PR considering there should be several commits anyway.
